llm/llama.go

```diff
@@ -544,11 +544,18 @@ func (llm *llama) Predict(ctx context.Context, prevContext []int, prompt string,
 		Stop:             llm.Stop,
 	}
 
-	data, err := json.Marshal(predReq)
-	if err != nil {
-		return fmt.Errorf("error marshaling data: %v", err)
+	// Handling JSON marshaling with special characters unescaped.
+	buf := &bytes.Buffer{}
+	enc := json.NewEncoder(buf)
+	enc.SetEscapeHTML(false)
+
+	if err := enc.Encode(predReq); err != nil {
+		return fmt.Errorf("failed to marshal data: %v", err)
 	}
 
+	// Remove trailing newline added by Encode.
+	data := buf.Bytes()[:buf.Len()-1]
+
 	req, err := http.NewRequestWithContext(ctx, http.MethodPost, endpoint, bytes.NewBuffer(data))
```

Comment:
Leaving the newline allows this to take `buf` instead of creating a new buffer

Reply:
```json
{"prompt":"<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\nHi<|im_end|>\n<|im_start|>assistant\n","stream":true,"n_predict":500,"n_keep":20,"temperature":0.8,"top_k":40,"top_p":0.9,"tfs_z":1,"typical_p":1,"repeat_last_n":64,"repeat_penalty":1.1,"presence_penalty":0,"frequency_penalty":0,"mirostat":0,"mirostat_tau":5,"mirostat_eta":0.1,"penalize_nl":true,"seed":-1,"stop":["<|im_end|>","<|im_start|>"]}
```

This is a `print(buffer.Bytes())` after the changes.